jcli/src/jcli_app/utils/debug_flag.rs

```diff
@@ -9,10 +11,24 @@ pub struct DebugFlag {
 }
 
 impl DebugFlag {
-    pub fn debug_writer(&self) -> Option<impl std::io::Write> {
-        match self.debug {
-            true => Some(std::io::stderr()),
-            false => None,
+    pub fn write_request(&self, request: &Request, body: &RestApiRequestBody) {
+        if !self.debug {
+            return;
+        }
+        eprintln!("{:#?}", request);
+        if body.has_body() {
+            eprintln!("Request body:\n{}", body)
+        }
+    }
+
+    pub fn write_response(&self, response: &RestApiResponse) {
```

Comment:
Odd that `RestApiResponse` contains the http response member, but `write_request` has to be passed disjoint request and body.

Reply:
Yes, it's an oddity coming from the fact that when body is passed to a `Request`, it's lost forever. All you can do is get the `Body`, which has no access to underlying data and that's by design. In order to do anything with the body data it must be passed separately before it's added to a `Request`.

Reply:
I can wrap these 2 values with a `RestApiRequest`, but I think that we won't gain much. A few inner functions will be getting 1 argument less, but the new structure won't have any logic to provide, it will be nothing more than a tuple.